src/frontend/js/hooks/useCourseRunOrder/index.tsx

```diff
@@ -0,0 +1,25 @@
+import { isJoanieEnabled } from 'api/joanie';
+import { extractResourceMetadata } from 'api/lms/joanie';
+import { useSession } from 'contexts/SessionContext';
+import { useOrders } from 'hooks/useOrders';
+import { CourseRun } from 'types';
+import { CourseLight } from 'types/Joanie';
+
+const useCourseRunOrder = (courseRun: CourseRun, course?: CourseLight) => {
+  const { user } = useSession();
+  const resourceLinkResources = extractResourceMetadata(courseRun.resource_link);
```

Comment:
If I'm right you should retrieve the course code here ?

Reply:
It's look like we can have both (see joanie's [CourseProductRelationViewSet](https://github.com/openfun/joanie/blob/main/src/backend/joanie/core/api/client/__init__.py#L104-L116)).
But you're right, the one used when we sync the product as a richie's course run is the code (see joanie's [CourseProductRelation](https://github.com/openfun/joanie/blob/main/src/backend/joanie/core/models/courses.py#L743-L756) model)

I would not say that i'm 100% `extractResourceMetadata` will return a code but now that i know it's the case here, i could remove this code property chain.

Reply:
```shell
$ git grep 'courses/.*code.*/products' | wc -l
40
$ git grep 'courses/.*id.*/products' | wc -l
27
```

Reply:
You're right, it looks risky to `extractResourceMetadata` to get course code.